src/required_resources.cpp

```diff
@@ -31,40 +31,36 @@ int RequiredResources::create_maps_impl(BPFtrace &bpftrace, bool fake)
   auto is_invalid_map = [=](int a) -> uint8_t {
     return (!fake && a < 0) ? 1 : 0;
   };
-  for (auto &map_val : map_vals) {
-    std::string map_name = map_val.first;
-    SizedType type = map_val.second;
+  for (auto &map_info : maps_info) {
+    std::string map_name = map_info.first;
+    SizedType type = map_info.second.value_type;
 
-    auto search_args = map_keys.find(map_name);
-    if (search_args == map_keys.end())
-      LOG(FATAL) << "map key \"" << map_name << "\" not found";
-
-    auto &key = search_args->second;
+    auto &key = map_info.second.key;
     auto max_map_keys = bpftrace.config_.get(ConfigKeyInt::max_map_keys);
 
     if (type.IsLhistTy()) {
-      auto args = lhist_args.find(map_name);
-      if (args == lhist_args.end())
+      const auto &args = map_info.second.lhist_args;
+      if (!args.has_value())
         LOG(FATAL) << "map arg \"" << map_name << "\" not found";
```

Comment:
Unrelated, but wondering if we should always use `LOG(FATAL)` instead of exceptions, since they serve similar purposes as documented in our coding guidelines. Would be an easy codemod

Reply:
This is something I'm often confused about. In some places, we use `LOG(FATAL)`, in others we raise `std::runtime_exception`. It'd be nice to unify this, perhaps we can discuss on the next office hours.

Reply:
Sounds good!